manifests/init.pp

```diff
@@ -536,6 +538,20 @@
     }
   }
 
+  case type3x($ssh_key_export) {
```

Comment:
While this will be shorter and will likely make the spec tests a bit simpler, if someone using the module makes a typo or otherwise enters an invalid value, it's going to throw either "Puppet::Error: str2bool(): Unknown type of boolean given" or "Puppet::Error: 'theTypodString' is not a boolean.", which doesn't really explain the issue to the user - it just looks like the module is broken internally. I would argue in favor of having the tests be slightly more complex in order for the error messages to be more meaningful to a new user of the module who might not understand the internals of it (and frankly, shouldn't have to to debug a compilation error).